src/backends/torch/torchdataaug.cc

```diff
@@ -27,47 +27,127 @@ namespace dd
   void TorchImgRandAugCV::augment(cv::Mat &src)
   {
     // apply augmentation
-    if (_mirror)
-      applyMirror(src);
-    if (_rotate)
-      applyRotate(src);
-    if (_geometry)
-      applyGeometry(src);
+    applyMirror(src);
+    applyRotate(src);
+    applyGeometry(src, _geometry_params);
 
     // should be last, in this order
-    if (_cutout > 0.0)
-      applyCutout(src);
-    if (_crop_size > 0)
-      applyCrop(src);
+    applyCutout(src, _cutout_params); // TODO: _img_width / _img_height could
+                                      // be wrong after other transforms
+    applyCrop(src, _crop_params);
   }
 
-  void TorchImgRandAugCV::applyMirror(cv::Mat &src)
+  void
+  TorchImgRandAugCV::augment_with_bbox(cv::Mat &src,
+                                       std::vector<torch::Tensor> &targets)
   {
+    // debug
+    // static int ii = 0;
+    // debug
+
+    torch::Tensor t = targets[0];
+    int nbbox = t.size(0);
+    std::vector<std::vector<float>> bboxes;
+    for (int bb = 0; bb < nbbox; ++bb)
+      {
+        std::vector<float> bbox;
+        for (int d = 0; d < 4; ++d)
+          {
+            bbox.push_back(t[bb][d].item<float>());
+          }
+        bboxes.push_back(bbox); // add (xmin, ymin, xmax, ymax)
+      }
+
+    bool mirror = applyMirror(src);
+    if (mirror)
+      {
+        applyMirrorBBox(bboxes, static_cast<float>(src.cols));
+      }
+    int rot = applyRotate(src);
+    if (rot > 0)
+      {
+        applyRotateBBox(bboxes, static_cast<float>(src.cols),
+                        static_cast<float>(src.rows), rot);
+      }
+    // XXX: no cutout with bboxes (yet)
+    GeometryParams geoparams = _geometry_params;
+    applyGeometry(src, geoparams, true);
+    if (!geoparams._lambda.empty())
+      {
+        // geometry on bboxes
+        applyGeometryBBox(bboxes, geoparams, src.cols, src.rows);
+      }
+
+    // replacing the initial bboxes with the transformed ones.
+    nbbox = bboxes.size();
+    for (int bb = 0; bb < nbbox; ++bb)
+      {
+        for (int d = 0; d < 4; ++d)
+          {
+            t[bb][d] = bboxes.at(bb).at(d);
+          }
+
+        // debug
+        /*cv::Mat src_bb = src.clone();
+        cv::Rect r(bboxes[bb][0], bboxes[bb][1], bboxes[bb][2] - bboxes[bb][0],
+                   bboxes[bb][3] - bboxes[bb][1]);
+        cv::rectangle(src_bb, r, cv::Scalar(255, 0, 0), 1, 8, 0);
+        cv::imwrite("/tmp/test_aug_" + std::to_string(ii)
+                        + ".png",
+                    src_bb);
+                    ++ii;*/
+        // debug
```

Comment:
It could be useful to put this in a function somewhere, in case we need it to debug data augmentation later

Reply:
done